gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/UserServiceImpl.java

```diff
@@ -467,4 +501,23 @@ private Completable checkRoles(List<String> roles) {
                 }).ignoreElement();
     }
 
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    private void createPasswordHistory(Domain domain, io.gravitee.am.model.User user, String rawPassword, io.gravitee.am.identityprovider.api.User principal) {
+        passwordHistoryService
+                .addPasswordToHistory(DOMAIN, domain.getId(), user, rawPassword , principal, domain.getPasswordSettings())
+                .subscribe(passwordHistory -> LOGGER.debug("Created password history for user with ID {}", user),
+                           throwable -> LOGGER.debug("Failed to create password history", throwable));
+    }
+
+
+    @SuppressWarnings({"ResultOfMethodCallIgnored", "ReactiveStreamsUnusedPublisher"})
+    private Single<io.gravitee.am.identityprovider.api.User> createPasswordHistory(
+            Domain domain, io.gravitee.am.model.User user,
+            String rawPassword,
+            io.gravitee.am.identityprovider.api.User principal, Function<PasswordHistory, SingleSource<? extends io.gravitee.am.identityprovider.api.User>> function) {
```

Comment:
Open discussion:

I'm not a big fan of passing a reactive flow as parameter here. I do prefer a createPasswordHistory method that returns a Single/Maybe and continue the processing outside the method.
Something like :
```
createPasswordHistory()
  .map/flatMap(ph -> userProvider.update(user))
  .flapMap(user -> userProvider.updatePassword)
```
wdyt ?